packages/fs-detectors/src/detect-builders.ts

```diff
@@ -305,6 +305,26 @@ export async function detectBuilders(
     };
   }
 
+  // Exclude the middleware builder for Next.js apps since @vercel/next
+  // will build middlewares.
+  //
+  // While maybeGetApiBuilder() excludes the middleware builder, however,
+  // we need to check if it's a Next.js app here again for the case where
+  // `projectSettings.framework == null`.
+  if (
+    framework === null &&
+    frontendBuilder?.use === '@vercel/next' &&
+    projectSettings.createdAt &&
+    projectSettings.createdAt < Date.parse('2020-03-01') &&
```

Comment:
I don’t think you need to check the date here since it’s already checked when frontendBuilder is assigned